lib/Sema/CodeSynthesis.cpp

```diff
@@ -2551,7 +2563,7 @@ static void maybeAddMemberwiseDefaultArg(ParamDecl *arg, VarDecl *var,
   bool isNilInitialized =
     var->getAttrs().hasAttribute<LazyAttr>() ||
     (!isExplicitlyInitialized && isDefaultInitializable &&
-     var->getValueInterfaceType()->getAnyNominal() == ctx.getOptionalDecl() &&
+     isNilOrTupleOfNils(var->getValueInterfaceType(), ctx) &&
      !var->getAttachedPropertyWrapperTypeInfo(0).defaultInit);
   if (isNilInitialized) {
     arg->setDefaultArgumentKind(DefaultArgumentKind::NilLiteral);
```

Comment:
Does SILGen know how to lower a call to a DefaultArgumentKind::NilLiteral whose type is a tuple? You should add a SILGen test for calling a memberwise init with a tuple of Optionals. I think this will fail right now because getCallerDefaultArg() needs to build the right nesting of TupleExprs and NIlLiteralExprs in the DefaultArgumentKind::NilLiteral case. This should share code with TypeChecker::buildDefaultInitializer().

Reply:
Since I'm not going forward with 'NilLiteral could also mean tuple of nils', this wasn't an issue. I added a SILGen test to make sure we did the right thing.